src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -729,4 +729,11 @@ object messages {
            | - auxiliary constructors specify the implicit value
            |"""
   }
+
+  case class NoClassOrObjectAfterCase()(implicit ctx: Context)
+  extends Message(27) {
+    val kind = "Syntax"
+    val msg = hl"Only ${"class"} or ${"object"} are allowed after ${"case"}."
+    val explanation = ""
```

Comment:
I think the explanation should be something explaining to the user why (s)he can't create a `case trait`. Perhaps you don't need to explain case classes/objects but the user should be given some alternatives to fix the error.

For example: (s)he could use `abstract case class`